### PR TITLE
Prep documentation for component catalog

### PR DESCRIPTION
**Pull request by margaree**
This PR does the following:
- Adds jsdoc comments on properties to the component files for htmleditor
- Adds analyzer step to release in order to include custom-elements for easily pulling property & event information into component catalog
- Updates README content to contain design's recommendation of content and hides pieces from component catalog that aren't needed (e.g., installation instructions)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,42 +1,41 @@
-# d2l-htmleditor
+# HTML Editor
 
+<!-- docs: start hidden content -->
```
Comment (margaree) [inline README.md:3]: The comments throughout are used by the component catalog page generation tooling and cannot be seen by viewers of the README

File: README.md [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
 [![NPM version](https://img.shields.io/npm/v/@brightspace-ui/htmleditor.svg)](https://www.npmjs.org/package/@brightspace-ui/htmleditor)
+<!-- docs: end hidden content -->
 
-An HTML editor that integrates with Brightspace.
+An HTML Editor component used to author rich HTML documents. It offers three authoring modes for different contexts.
 
-## Installation
```
Comment (margaree) [inline README.md:7]: This was just moved towards the bottom of the document

File: README.md [hunk 3]
```diff
@@ -1,42 +1,41 @@
...
-```
+<!-- docs: start donts -->
+* Don’t use when multiple editors appear in close proximity; consider Inline mode instead
+* Don’t shrink the Full editor down to fit in tight spaces, such as small sidebars
+* Don’t use when rich text formatting is unnecessary or certain formatting options are disallowed
+<!-- docs: end donts -->
+<!-- docs: end best practices -->
 
-Types of editors (toolbar features):
+<!-- docs: demo live name:d2l-htmleditor disableIframe:true -->
```
Comment (margaree) [inline README.md:29]: FYI @GeorgePlukov `disableIframe` is being proposed as a part of https://github.com/BrightspaceUI/documentation/pull/345

**Review by grant-cleary — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -1,42 +1,41 @@
...
-HTML document (including `head` & `body`):
-```html
-<d2l-htmleditor html="..." label="..." full-page></d2l-htmleditor>
-```
+<!-- docs: start donts -->
+* Don’t use when multiple editors appear in close proximity; consider Inline mode instead
+* Don’t shrink the Full editor down to fit in tight spaces, such as small sidebars
+* Don’t use when rich text formatting is unnecessary or certain formatting options are disallowed
+<!-- docs: end donts -->
+<!-- docs: end best practices -->
```
Comment (grant-cleary) [inline README.md:27]: I don't have any opinion on this either way, but is it intentional that we only list these best practices for the full mode?

Comment (margaree) [reply README.md:27]: It is, yup - I believe they were written to be more specific to full mode, and then the other two modes have their own best practices as well

Comment (grant-cleary) [reply README.md:27]: Oh, I missed down below that the other set of "Best Practices" is specific to inline mode. 🤦

**Comment by grant-cleary** [comment]
LGTM. Really excited to see the component catalog when it's ready! 😃

**Pull request merged**